### PR TITLE
#4917 fix: initialize external service at init

### LINKED ISSUE #4917: Set external service on AccountServer init
> Maybe it's possible to check `isCustomerUrlFesUsed(this.acctEmail)` on `AccountServer` init and set service url there, so we don't need to run `getExternalService` before each request?

_Originally posted by @sosnovsky in https://github.com/FlowCrypt/flowcrypt-browser/pull/4907#discussion_r1090509963_

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR initialized external service at init in AccountServer

close #4917 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/api/account-server.ts [hunk 1]
```diff
@@ -20,19 +20,21 @@ export type UploadedMessageData = {
...
-    this.potentialCustomerUrlFes = new ExternalService(this.acctEmail);
-    this.sharedTenantFes = new ExternalService(this.acctEmail);
-    this.sharedTenantFes.url = SHARED_TENANT_API_HOST;
+    this.externalService = new ExternalService(this.acctEmail);
   }
 
+  public initialize = async () => {
+    if (!(await isCustomerUrlFesUsed(this.acctEmail))) {
+      this.externalService.url = SHARED_TENANT_API_HOST;
+    }
```
Comment (sosnovsky) [inline extension/js/common/api/account-server.ts:33]: Can we move this check to `constructor`? So we won't need to run it separately on each `AccountServer` initialisation. 

Then this code
```ts
const acctServer = new AccountServer(acctEmail);
await acctServer.initialize();
await acctServer.fetchAndSaveClientConfiguration();
```

Can be just 
```ts
await new AccountServer(acctEmail).fetchAndSaveClientConfiguration();
```
as it was before

Comment (ioanmo226) [reply extension/js/common/api/account-server.ts:33]: But as you know `initialize` is async functino and can't be placed in constructor.

Comment (sosnovsky) [reply extension/js/common/api/account-server.ts:33]: What about using static `init` function, like this
```ts
class AccountServer  {
  public static init = async (acctEmail: string) => {
    const acctServer = new AccountServer(acctEmail);
    await acctServer.initialize();
    return acctServer
  };
```

And then used this way
```ts
await AccountServer.init(acctEmail).fetchAndSaveClientConfiguration();
```

In this case we won't forget to run `initialize` for each `AccountServer` init

Comment (ioanmo226) [reply extension/js/common/api/account-server.ts:33]: It sounds good when AccountServier init was called from async funtion.
But in several places, we init AccountServer in constructor (like ComposeView consturctor and accountserver was used from other view modules).
For those places, we still need to call initalize in render function or so

Comment (sosnovsky) [reply extension/js/common/api/account-server.ts:39]: Got it 👍 Looks good now

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**